### PR TITLE
Add datadog logging for entity store usages

### PR DESCRIPTION
**Pull request by peppy**
Bare minimum logging to track the usage of entity stores (and hopefully help track down issues)

This is already deployed live and has helped with some basic understanding. Have confirmed that everything is outputting as expected, but would appreciate a quick one-over on the calls from a correctness perspective.

### CODE REVIEW

File: osu.Server.Spectator/Hubs/EntityStore.cs [hunk 1]
```diff
@@ -62,6 +70,7 @@ public async Task<ItemUsage<T>> GetForUse(long id, bool createOnMissing = false)
                     throw new KeyNotFoundException($"Attempted to get untracked entity {typeof(T)} id {id}");
```
Comment (bdach) [inline osu.Server.Spectator/Hubs/EntityStore.cs:70]: this unlikely branch could probably also use an increment if we're tracking missing keys, similar/same as line 48

**Comment by bdach** [comment]
Have looked at what the datadog operations are doing via inspecting decompiled IL in rider and it seems to do the right thing (events are enqueued onto an asynchronous worker which flushes every [1,100] ms. So I don't think adding this logging should be increasing lock contention much.

**Comment by peppy** [comment]
Yep, the overhead is zero (and even if it wasn't the target server is localhost and it's using UDP). We already use this tens of thousands of times per second in bancho, web-10 and osu-web.

**Review by bdach — APPROVED** [review]
seems logical

**Pull request merged**